authoriser/src/main/java/uk/nhs/digital/docstore/authoriser/handlers/VerifyOrganisationHandler.java

```diff
@@ -0,0 +1,129 @@
+package uk.nhs.digital.docstore.authoriser.handlers;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.nimbusds.oauth2.sdk.id.Subject;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.nhs.digital.docstore.authoriser.Utils;
+import uk.nhs.digital.docstore.authoriser.enums.HttpStatus;
+import uk.nhs.digital.docstore.authoriser.repository.DynamoDBSessionStore;
+import uk.nhs.digital.docstore.authoriser.repository.SessionStore;
+import uk.nhs.digital.docstore.authoriser.requestEvents.OrganisationRequestEvent;
+
+public class VerifyOrganisationHandler extends BaseAuthRequestHandler
+        implements RequestHandler<OrganisationRequestEvent, APIGatewayProxyResponseEvent> {
+    public static final Logger LOGGER = LoggerFactory.getLogger(VerifyOrganisationHandler.class);
+    private final SessionStore sessionStore;
+    private Clock clock = Clock.systemUTC();
+    private static final String ODS_CODE_PARAM_KEY = "odsCode";
+
+    @SuppressWarnings("unused")
+    public VerifyOrganisationHandler() {
+        this(new DynamoDBSessionStore(createDynamoDbMapper()));
+    }
+
+    public VerifyOrganisationHandler(SessionStore sessionStore, Clock clock) {
+        this.sessionStore = sessionStore;
+        this.clock = clock;
+    }
+
+    public VerifyOrganisationHandler(SessionStore sessionStore) {
+        this.sessionStore = sessionStore;
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            OrganisationRequestEvent input, Context context) {
+        LOGGER.debug("Request event: " + input);
+        LOGGER.debug("QSPs: " + input.getQueryStringParameters().toString());
+
+        var odsCode = Optional.ofNullable(input.getQueryStringParameters().get(ODS_CODE_PARAM_KEY));
+        var sessionId = input.getSessionId();
+        var subjectClaim = input.getSubjectClaim();
+
+        var error = false;
+
+        if (odsCode.isEmpty()) {
+            error = true;
+            LOGGER.error("ODS code is missing");
+        }
+
+        if (sessionId.isEmpty()) {
+            error = true;
+            LOGGER.error("SessionId is missing");
+        }
+
+        if (subjectClaim.isEmpty()) {
+            error = true;
+            LOGGER.error("SubjectClaim is missing");
+        }
+
+        if (error) {
+            return orgHandlerError(HttpStatus.BAD_REQUEST.code);
```

Comment:
We have discussed error handling with Dawn and Ifey with relation to users and have decided to show a generic error page if an error like this occurs. This is because it's unlikely a user would have the right technical knowledge to correct this issue themselves if it occured. We have a seperate ticket to add auditing into the code so we can pick up issues like this in Splunk which can then send up approriate alerts.